lib/Dialect/TTNN/Transforms/TTNNWorkarounds.cpp

```diff
@@ -279,11 +331,22 @@ static bool propagateDpsInitChangesToDpsResults(wa::TTNNWorkaroundInterface &op,
     OpOperand *dpsInit = dpsOp.getDpsInitOperand(dpsInitIndex);
     OpResult tiedDpsResult = dpsOp.getTiedOpResult(dpsInit);
 
-    // If the DPS destination is changed, update the DPS result as well.
+    // If the DPS destination is changed   t as well.
     if (tiedDpsResult.getType() != dpsInit->get().getType()) {
       modified = true;
+      // Preserve the previous data type of the DPS result.
+      DataType previousOutputDataType =
+          getLayoutAttrFromValue(tiedDpsResult).getDataType();
+
+      // Update the DPS result type with the new tensor layout, data type,
+      // buffer type and memory layout.
       rewriter.modifyOpInPlace(
           op, [&]() { tiedDpsResult.setType(dpsInit->get().getType()); });
+
+      // If the data type of the output result has changed as a result of a data
+      // type workaround, cast the data type back to the previous data type.
+      castOutputDataTypeBack(op, rewriter, previousOutputDataType,
+                             tiedDpsResult);
     }
```

Comment:
If I understand this correctly, here you change current op result if it doesn't match dps operand, right? Would it make sense to cast input operand to op result type instead of opposite?

Reply:
Actually, with this PR you rollback the result of output workaround after it's applied on line 307. Then you update all it's users with to layout op which should have correct data type, so check on line 335 should always be false right?